### ISSUE TITLE
How to trigger an eventClick from outside the calendar?

### ISSUE THREAD

**Issue by godshades** [fullcalendar/fullcalendar#3328]
I have a list of the day. When I click to this day I want fire eventClick same as I click on calendar. Can someone help me

**Comment by harish0000** [comment]
@godshades you can use use the eventClick callback function 

```
eventClick:function(event,jsEvent,view){
console.log(event.id);
console.log(event.title);
}
```

Hope the above code resolves your issue

**Comment by godshades** [comment]
can u help me explain more, I need write click function inside eventClick right ?
I try this code but not work:
eventClick:function(event,jsEvent,view){
 $(".list-days").click(function(event) {
        openEvent(event);
      });
}

**Comment by harish0000** [comment]
@godshades  i didn't understand what exactly your requirements are ? from the above comment i understand you want to open the event on click of event ,Right ?

**Comment by godshades** [comment]
@harish0000 no, I have a list of event display like:
<a>Event1</a>
<a>Event2</a>
...
And a calendar have event draw on this:
<div id='calendar'></div>

When I click on Event1 in calendar it displays information about Event1
Now I want when I click on link Event1  of list, it do the same as I click on Event1 in calendar
Sr for bad English

**Comment by harish0000** [comment]
@godshades ahh i got it , you can do one thing give a class and an id attribute to your outside events ,so when u click on outside event get the id  attribute of that event and then trigger a click on the event present in the calendar (you can find the calendar event by id you got from outside event). 

$(document).on("click", ".outsideEventClass", function() {
      get the id;
      trigger a click event for full calendar (find the div containing the event id)
});

**Comment by godshades** [comment]
i try: 
$(".eventClass").click(function() {
    var eventId = $(this).attr('id');
    var events = $('#calendar').fullCalendar('clientEvents',eventId);
    if (events.length > 0) {
        $("#calendar").fullCalendar.trigger('eventClick', events[0]);
    }
});
and have this error: Uncaught TypeError: $(...).fullCalendar.trigger

**Comment by godshades** [comment]
@arshaw why close +_+ I still don't resolve my problem

**Comment by harish0000** [comment]
@godshades i am still thinking how full calendar manages the id's of events ,also there is no id attribute in any div of calendar. @arshaw can you help you in this.

**Comment by godshades** [comment]
@harish0000 I use eventRender to add id for div of calendar:
eventRender: function(event, element) {
       element.find('.fc-content').attr("id", "event-" + event.id);
    }

**Comment by harish0000** [comment]
@godshades if you have added the id attribute to event div ,then its easy you can trigger the click event

```
$(".eventClass").click(function() {
var eventId = $(this).attr('id');
$('.fc-content[event-"' + eventId + '"]').trigger('click');
});
```

**Comment by godshades** [comment]
@harish0000 thx, I'm done with that, but code is:
$('#' + eventId + '.fc-content ').trigger('click') 
:D

**Comment by Rajdipkhavad** [comment]
it's not working when another month is open in calendar

**Comment by magicmb** [comment]
> @harish0000 I use eventRender to add id for div of calendar:
> eventRender: function(event, element) {
> element.find('.fc-content').attr("id", "event-" + event.id);
> }

**Comment by magicmb** [comment]
I too was struggling for a way to fire a eventClick from outside fullCalendar and found the answers in this thread. 

However it took me a while to get my head around it and I wanted to share my own summary in case it helps others who stumble upon this thread

​
The first stumbling block as @godshades rightly pointed out was figuring out how to target specific events when event id attibutes were not in use on the calendar items

> @godshades i am still thinking how full calendar manages the id's of events ,also there is no id attribute in any div of calendar. @arshaw can you help you in this.


but @harish0000 reply pointed me in the right direction


> @harish0000 I use eventRender to add id for div of calendar:
> eventRender: function(event, element) {
> element.find('.fc-content').attr("id", "event-" + event.id);
> }

although my simplified implementation worked just as well

```
eventRender: function(event, element, view) {
          element.attr( 'id', event.id );
          ...
```

​
Now **providing** the outside **fullCalendar trigger knows about the event ID**, you can fire the eventClick by calling this from your own **on click** function similar to what both @harish0000 and @godshades suggested

> tweaked @harish0000 solution:
$('.fc-list-item[id="' + eventId + '"]').trigger('click');

or

> @godshades solution:
$('#' + eventId + '.fc-list-item').trigger('click');

​
Both worked for me and in my own implementation I use the **eventRender** process to dynamically create a "subset" list of events that resides **outside fullCalendar** and this process also adds the ID of the event to each entry for easy retrieval by an *on click** eventListener function that I personally use to fire the above trigger.

`$list.on( 'click', '.list-item', function(e) { `

​
The **only other issue** I struggled with was how to **retrieve the ID** (eventId) to **pass into trigger** used inside the **.on Click** eventListener function. In my particular case I manage to overcome this by grabbing the **outerHTML** of the first DIV on the **.list-item** 

`var eFirstDiv = e.currentTarget.firstChild;  `

and then taking a substring of this to yield my **eventId** which tied it all nicely together

​
In summary this is what I did

- **Add ID** to **Calendar** Items
- **Add ID** to **HTML** List Items
- Create **on Click** event Listener which **retrieves HTML ID** and **triggers eventClick**

**Comment by hotcakedev628** [comment]
@harish0000 @godshades I am using fullcalendar v5.5 in my CakePHP application.

Everything works fine, but the `eventClick` seems not working.

```
document.addEventListener('DOMContentLoaded', function() {
    var calendarAgendaEl = document.getElementById('agenda-calendar');

    var calendarAgenda = new FullCalendar.Calendar(calendarAgendaEl, {
        headerToolbar: {
            left: 'prevYear,prev,next,nextYear today',
            center: 'title',
            right: 'dayGridMonth,dayGridWeek,dayGridDay'
        },
        dayMaxEvents: true,
        eventSources: [{
            method: 'POST',
            url: '/calendar/get_all',

            failure: function() {
                document.getElementById('script-warning').style.display = 'block'
            }
        }],
        eventSourceSuccess: function(content, xhr) {
            var events = [];

            content.events.value.map(event => {
                events.push({
                    id: event.id,
                    allDay: event.isAllDay,
                    title: event.subject,
                    start:event.start.dateTime,
                    end: event.end.dateTime,
                    extendedProps: {
                        description: event.bodyPreview,
                    }
                    // className: ['box']
                })
            })

            return events;
        },
        eventClick: function(info) {
            alert('Clicked an event!!!');
            info.jsEvent.preventDefault();
        }
    });

    calendarAgenda.render();
})
```
The code is from agenda.js file and the agenda-calendar element is in the view template php file.
`<div id='agenda-calendar'></div>`

I am using cdn to import jQuery like the following, and not using jQuery UI.
`<script type="text/javascript" src="https://cdn.jsdelivr.net/jquery/latest/jquery.min.js"></script>`

What is the reason why nothing happens if I click an event?
